### PR TITLE
Change DLK test runner from nose2 to pytest

### LINKED ISSUE #602: Make dlk test executable with pytest command
Currently, we are running the dlk test with the setup.py command.

```
PYTHONPATH=python/dlk python setup.py test
```

We want to make dlk test executable with pytest command.

```
pytest
```

### PR DESCRIPTION
**Pull request by hadusam**
Close #602

### CODE REVIEW

File: dlk/tests/test_code_generation.py [hunk 1]
```diff
@@ -416,7 +417,7 @@ def codegen_cpu(self,
 class TestCodeGenerationX8664(TestCodeGenerationBase):
     """Test class for code generation testing."""
 
-    @params(*get_configurations_x86_64())
+    @parameterized.expand(get_configurations_x86_64())
```
Comment (suttang) [inline dlk/tests/test_code_generation.py]: Can't use `@pytest.mark.parametrize`?

Comment (hadusam) [reply dlk/tests/test_code_generation.py]: I've tried it first, but unfortunately, `@pytest.mark.parametrize` does not work in `unittest.TestCase` subclasses.
https://docs.pytest.org/en/latest/unittest.html#pytest-features-in-unittest-testcase-subclasses
`import parameterized` is an easy way to remove nose2 dependency without large structural changes(e.g. removing inheritance of unittest.TestCase).

Comment (suttang) [reply dlk/tests/test_code_generation.py]: OH... Thank you for your explanation!!

**Comment by tkng** [comment]
LGTM, please merge after the conflict is resolved.

**Pull request merged**